notebooks/03_evaluate/comparison_movielens.ipynb

```diff
@@ -0,0 +1,982 @@
+    "    \"sar_single_node\": lambda model, test, train: recommend_k_sar_single_node(model, test, train),\n",
+    "    \"svd\": lambda model, test, train: recommend_k_svd(model, test, train),\n",
+    "    \"fastai\": lambda model, test, train: recommend_k_fastai(model, test, train),\n",
+    "    \"ncf\": lambda model, test, train: recommend_k_ncf(model, test, train),\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Metrics"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def rating_metrics_pyspark(test, predictions):\n",
+    "    rating_eval = SparkRatingEvaluation(test, \n",
+    "                                        predictions, \n",
+    "                                        col_user=USER_COL, \n",
+    "                                        col_item=ITEM_COL, \n",
+    "                                        col_rating=RATING_COL, \n",
+    "                                        col_prediction=PREDICTION_COL)\n",
+    "    return {\n",
+    "        \"RMSE\": rating_eval.rmse(),\n",
+    "        \"MAE\": rating_eval.mae(),\n",
+    "        \"R2\": rating_eval.exp_var(),\n",
+    "        \"Explained Variance\": rating_eval.rsquared()\n",
+    "    }\n",
+    "    \n",
+    "    \n",
+    "def ranking_metrics_pyspark(test, predictions, k=10):\n",
+    "    rank_eval = SparkRankingEvaluation(test, \n",
+    "                                       predictions, \n",
+    "                                       k=k, \n",
+    "                                       col_user=USER_COL, \n",
+    "                                       col_item=ITEM_COL, \n",
+    "                                       col_rating=RATING_COL, \n",
+    "                                       col_prediction=PREDICTION_COL, \n",
+    "                                       relevancy_method=\"top_k\")\n",
+    "    return {\n",
+    "        \"MAP\": rank_eval.map_at_k(),\n",
+    "        \"nDCG@k\": rank_eval.ndcg_at_k(),\n",
+    "        \"Precision@k\": rank_eval.precision_at_k(),\n",
+    "        \"Recall@k\": rank_eval.recall_at_k()\n",
+    "    }\n",
+    "    \n",
+    "    \n",
+    "def rating_metrics_python(test, predictions):\n",
+    "    cols = {\n",
+    "        \"col_user\": USER_COL, \n",
+    "        \"col_item\": ITEM_COL, \n",
+    "        \"col_rating\": RATING_COL, \n",
+    "        \"col_prediction\": PREDICTION_COL\n",
+    "    }\n",
+    "    return {\n",
+    "        \"RMSE\": rmse(test, predictions, **cols),\n",
+    "        \"MAE\": mae(test, predictions, **cols),\n",
+    "        \"R2\": rsquared(test, predictions, **cols),\n",
+    "        \"Explained Variance\": exp_var(test, predictions, **cols)\n",
+    "    }\n",
+    "    \n",
+    "    \n",
+    "def ranking_metrics_python(test, predictions, k=10):\n",
+    "    cols = {\n",
+    "        \"col_user\": USER_COL, \n",
```

Comment:
Similarly here, you could use `eval(metric)` where metric is the name of the metric.